### ISSUE TITLE
Marker "inheritance" not consistent with pytest behaviour

### ISSUE THREAD

**Issue by ROpdebee** [pytest-dev/pytest-describe#36]
When markers are "stacked", pytest somehow keeps all markers, whereas pytest-describe throws away all but one instance of the markers. This has been reported in #23 for `parametrize`, but the same issue is applicable to all markers, also custom ones.

Here's an actual test case to exemplify it:
```python
import pytest

@pytest.fixture()
def get_marks(request):
    return [mark.args[0] for mark in request.node.iter_markers(name='my_mark')]

@pytest.mark.my_mark('foo')
class TestMark:

    def test_inherited_from_parent(self, get_marks):
        assert get_marks == ['foo']

    @pytest.mark.my_mark('bar')
    @pytest.mark.my_mark('baz')
    def test_all_marks_are_chained(self, get_marks):
        assert get_marks == ['baz', 'bar', 'foo']


@pytest.mark.my_mark('foo')
def describe_marks():
    def it_is_inherited_from_describe_block(get_marks):
        assert get_marks == ['foo']

    @pytest.mark.my_mark('bar')
    @pytest.mark.my_mark('baz')
    def all_marks_are_chained(get_marks):
        assert get_marks == ['baz', 'bar', 'foo']

```
The final test `describe_marks.all_marks_are_available` fails, with the actual value for `get_marks` being `['bar']`. They should instead be stacked.

Of course, this is a contrived example, but there's actual use cases for stacked/inherited markers, like the following:
```python
@pytest.mark.usefixtures('fixture_with_side_effects')
def describe_something():
    @pytest.mark.usefixtures('other_fixture_with_side_effects')
    def it_does_something():
         ...

    def it_does_something_else():
         # No need for 'other_fixture_with_side_effects' here
         ...
```

Python version: 3.8.2
pytest version: 6.1.2
pytest-describe version: latest master

**TL;DR:** Marker stacking/chaining/inheritance handling isn't consistent with pytest's builtin behaviour.

**Comment by ROpdebee** [comment]
Is there any specific reason why the defined markers are converted to a dict, which is then merged to remove duplicate keys? https://github.com/pytest-dev/pytest-describe/blob/d65c89dc7ad7637793df06afd3992601e3647823/pytest_describe/plugin.py#L64-L67
Removing that deduplication solves this issue, but it might break something else? In any case, I'm not sure whether that's the best solution, IMO it makes more sense to define the markers on the DescribeBlock itself, since that would be most consistent with pytest's internals.
